.environment-scripts/gnu_docker/variables.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# NCEPlibs arguments
+NCEPLIBS_PLATFORM=linux
+NCEPLIBS_COMPILER=gnu
+
+# ESMF arguments and environment variables
+ESMF_OS=Linux
+ESMF_COMPILER=gfortran
+ESMF_SITE=default
+ESMF_CC=
+
+# FMS environment variables
+FMS_CC=/usr/bin/mpicc
+FMS_FC=/usr/bin/mpif90
+FMS_LDFLAGS='-L/usr/lib'
+FMS_LOG_DRIVER_FLAGS='--comments'
+FMS_CPPFLAGS='-I/usr/include -Duse_LARGEFILE -DMAXFIELDMETHODS_=500 -DGFS_PHYS'
+FMS_FCFLAGS='-fcray-pointer -Waliasing -ffree-line-length-none -fno-range-check -fdefault-real-8 -fdefault-double-8 -fopenmp'
+
+# fv3gfs-fortran arguments
+FV3GFS_PLATFORM=gnu_docker
+
+# Python requirements environment variables
+MPI4PY_CC=/usr/bin/mpicc
+MPI4PY_MPICC=/usr/bin/mpicc
+
+# Python wrapper environment variables
+WRAPPER_CC='x86_64-linux-gnu-gcc -pthread'
```

Comment:
I'm still investigating what these need to be set to.  Ideally what we would set them to would allow us to reproduce the behavior of the current prognostic run image.  In that case neither `CC` or `LDSHARED` are defined in the build environment:

<details>
<p>

```
#18 34.47 x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/include/python3.8 -c lib/_wrapper.c -o build/temp.linux-x86_64-3.8/lib/_wrapper.o
#18 34.58 In file included from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1969,
#18 34.58                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarrayobject.h:12,
#18 34.58                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
#18 34.58                  from lib/_wrapper.c:757:
#18 34.58 /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
#18 34.58    17 | #warning "Using deprecated NumPy API, disable it with " \
#18 34.58       |  ^~~~~~~
#18 109.8 x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/lib/_wrapper.o -o /fv3net/external/fv3gfs-fortran/FV3/wrapper/fv3gfs/wrapper/_wrapper.cpython-38-x86_64-linux-gnu.so /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/coupler_lib.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/physics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/dynamics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/flagstruct_data.o /fv3net/external/fv3gfs-fortran/FV3/atmos_model.o /fv3net/external/fv3gfs-fortran/FV3/module_fv3_config.o -L/fv3net/external/fv3gfs-fortran/FV3 -lfv3core -lfv3io -lipd -lgfsphys -lstochastic_physics -lfv3cpl -lfv3coarse_graining -L/fv3net/external/fv3gfs-fortran/FV3/atmos_cubed_sphere -L/fv3net/external/fv3gfs-fortran/FV3/io -L/fv3net/external/fv3gfs-fortran/FV3/ipd -L/fv3net/external/fv3gfs-fortran/FV3/gfsphysics -L/fv3net/external/fv3gfs-fortran/FV3/../stochastic_physics -L/fv3net/external/fv3gfs-fortran/FV3/cpl -L/fv3net/external/fv3gfs-fortran/FV3/coarse_graining -L/usr/local/NCEPlibs/lib -lnemsio_d -lbacio_4 -lsp_v2.0.2_d -lw3emc_d -lw3nco_d -L/usr/local/esmf/lib -lesmf -lrt -lstdc++ -ldl -L/fv3net/external/fv3gfs-fortran/FMS/libFMS/.libs/ -fopenmp -L/usr/local/lib -lcallpy -lgfortran -lFMS -lesmf -lnetcdff -lnetcdf -llapack -lblas -lc -lrt -Wl,-Bsymbolic-functions -Wl,-z,relro -L/usr/lib/x86_64-linux-gnu -lmpichfort -lmpich -lmvec -lm
```

</p>
</details>

What I set these variables to here are my best guesses as to what they should be based on the above behavior.  If I set them like this, however, for some reason `mpicc` is used in the compile step, which succeeds, but is different than what is done currently in fv3net:

<details>
<p>

```
#17 37.08 running build_ext
#17 37.33 building 'fv3gfs.wrapper._wrapper' extension
#17 37.33 creating build
#17 37.33 creating build/temp.linux-x86_64-3.8
#17 37.33 creating build/temp.linux-x86_64-3.8/lib
#17 37.33 mpicc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/include/python3.8 -c lib/_wrapper.c -o build/temp.linux-x86_64-3.8/lib/_wrapper.o
#17 37.49 In file included from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1969,
#17 37.49                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarrayobject.h:12,
#17 37.49                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
#17 37.49                  from lib/_wrapper.c:757:
#17 37.49 /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
#17 37.49    17 | #warning "Using deprecated NumPy API, disable it with " \
#17 37.49       |  ^~~~~~~
#17 132.5 x86_64-linux-gnu-gcc -pthread -shared -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/lib/_wrapper.o -o /fv3net/external/fv3gfs-fortran/FV3/wrapper/fv3gfs/wrapper/_wrapper.cpython-38-x86_64-linux-gnu.so /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/coupler_lib.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/physics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/dynamics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/flagstruct_data.o /fv3net/external/fv3gfs-fortran/FV3/atmos_model.o /fv3net/external/fv3gfs-fortran/FV3/module_fv3_config.o -L/fv3net/external/fv3gfs-fortran/FV3 -lfv3core -lfv3io -lipd -lgfsphys -lstochastic_physics -lfv3cpl -lfv3coarse_graining -L/fv3net/external/fv3gfs-fortran/FV3/atmos_cubed_sphere -L/fv3net/external/fv3gfs-fortran/FV3/io -L/fv3net/external/fv3gfs-fortran/FV3/ipd -L/fv3net/external/fv3gfs-fortran/FV3/gfsphysics -L/fv3net/external/fv3gfs-fortran/FV3/../stochastic_physics -L/fv3net/external/fv3gfs-fortran/FV3/cpl -L/fv3net/external/fv3gfs-fortran/FV3/coarse_graining -L/usr/local/NCEPlibs/lib -lnemsio_d -lbacio_4 -lsp_v2.0.2_d -lw3emc_d -lw3nco_d -L/usr/local/esmf/lib -lesmf -lrt -lstdc++ -ldl -L/fv3net/external/fv3gfs-fortran/FMS/libFMS/.libs/ -fopenmp -L/usr/local/lib -lcallpy -lgfortran -lFMS -lesmf -lnetcdff -lnetcdf -llapack -lblas -lc -lrt -Wl,-Bsymbolic-functions -Wl,-z,relro -L/usr/lib/x86_64-linux-gnu -lmpichfort -lmpich -lmvec -lm
```

</p>
</details>

Leaving them blank leads to an error (basically it doesn't know what command to use in the linking step and for some reason it uses `mpicc` in the compile step):

<details>
<p>

```
#17 33.93 running build_ext
#17 34.15 building 'fv3gfs.wrapper._wrapper' extension
#17 34.15 creating build
#17 34.15 creating build/temp.linux-x86_64-3.8
#17 34.15 creating build/temp.linux-x86_64-3.8/lib
#17 34.15 mpicc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/local/lib/python3.8/dist-packages/numpy/core/include -I/usr/include/python3.8 -c lib/_wrapper.c -o build/temp.linux-x86_64-3.8/lib/_wrapper.o
#17 34.30 In file included from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1969,
#17 34.30                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/ndarrayobject.h:12,
#17 34.30                  from /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
#17 34.30                  from lib/_wrapper.c:757:
#17 34.30 /usr/local/lib/python3.8/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
#17 34.30    17 | #warning "Using deprecated NumPy API, disable it with " \
#17 34.30       |  ^~~~~~~
#17 112.1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.8/lib/_wrapper.o -o /fv3net/external/fv3gfs-fortran/FV3/wrapper/fv3gfs/wrapper/_wrapper.cpython-38-x86_64-linux-gnu.so /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/coupler_lib.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/physics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/dynamics_data.o /fv3net/external/fv3gfs-fortran/FV3/wrapper/lib/flagstruct_data.o /fv3net/external/fv3gfs-fortran/FV3/atmos_model.o /fv3net/external/fv3gfs-fortran/FV3/module_fv3_config.o -L/fv3net/external/fv3gfs-fortran/FV3 -lfv3core -lfv3io -lipd -lgfsphys -lstochastic_physics -lfv3cpl -lfv3coarse_graining -L/fv3net/external/fv3gfs-fortran/FV3/atmos_cubed_sphere -L/fv3net/external/fv3gfs-fortran/FV3/io -L/fv3net/external/fv3gfs-fortran/FV3/ipd -L/fv3net/external/fv3gfs-fortran/FV3/gfsphysics -L/fv3net/external/fv3gfs-fortran/FV3/../stochastic_physics -L/fv3net/external/fv3gfs-fortran/FV3/cpl -L/fv3net/external/fv3gfs-fortran/FV3/coarse_graining -L/usr/local/NCEPlibs/lib -lnemsio_d -lbacio_4 -lsp_v2.0.2_d -lw3emc_d -lw3nco_d -L/usr/local/esmf/lib -lesmf -lrt -lstdc++ -ldl -L/fv3net/external/fv3gfs-fortran/FMS/libFMS/.libs/ -fopenmp -L/usr/local/lib -lcallpy -lgfortran -lFMS -lesmf -lnetcdff -lnetcdf -llapack -lblas -lc -lrt -Wl,-Bsymbolic-functions -Wl,-z,relro -L/usr/lib/x86_64-linux-gnu -lmpichfort -lmpich -lmvec -lm
#17 112.1 error: command '-Wl,-Bsymbolic-functions' failed with exit status 1
```

</p>
</details>

So currently the only way I know how to reproduce the behavior of our current prognostic run image is to not define `CC` or `LDSHARED`.

Reply:
Resolved by ai2cm/fv3gfs-fortran#331.